src/fu-engine.c

```diff
@@ -2556,18 +2555,14 @@ fu_engine_install_release(FuEngine *self,
 				    progress,
 				    flags,
 				    feature_flags,
-				    &error_local)) {
-		if (g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_AC_POWER_REQUIRED) ||
-		    g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_BATTERY_LEVEL_TOO_LOW) ||
-		    g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_NEEDS_USER_ACTION) ||
-		    g_error_matches(error_local, FWUPD_ERROR, FWUPD_ERROR_BROKEN_SYSTEM)) {
-			fu_device_set_update_state(device_orig,
-						   FWUPD_UPDATE_STATE_FAILED_TRANSIENT);
-		} else {
+				    error)) {
+		FwupdUpdateState state = fu_device_get_update_state(device);
+		if (state != FWUPD_UPDATE_STATE_FAILED &&
+		    state != FWUPD_UPDATE_STATE_FAILED_TRANSIENT)
 			fu_device_set_update_state(device_orig, FWUPD_UPDATE_STATE_FAILED);
-		}
-		fu_device_set_update_error(device_orig, error_local->message);
-		g_propagate_error(error, g_steal_pointer(&error_local));
+		else
+			fu_device_set_update_state(device_orig, state);
+		fu_device_set_update_error(device_orig, (*error)->message);
```

Comment:
We have to use `error_local` -- the general GError ** promise is that it's either a `GError*` or NULL -- and if the latter (e.g. in tests, or in a gdb session) then this would explode.